error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1397,4 +1397,20 @@ Duration after(StepVerifier.LastStep step, Duration duration) {
       return step.verifyTimeout(duration);
     }
   }
+
+  /** Avoid collecting when verifying only the given element is in the given {@link Flux}. */
+  static final class VerifyOnlyElementInFlux<T> {
+    @BeforeTemplate
+    Duration before(Flux<T> flux, T object) {
+      return flux.collect(toImmutableList())
+          .as(StepVerifier::create)
+          .assertNext(list -> assertThat(list).containsExactly(object))
+          .verifyComplete();
+    }
+
+    @AfterTemplate
+    Duration after(Flux<T> flux, T object) {
+      return flux.as(StepVerifier::create).expectNext(object).verifyComplete();
```

Comment:
```suggestion
      return flux.as(StepVerifier::create).expectNext(object);
```
Besides the fact that the return types are now slightly different we can drop the `verifyComplete` part. 
Assuming the tests pass, we can do this rewrite as we test that the list literally contains one object. Strictly speaking this is not a behavior preserving change because of the updated return type. The `verifyComplete` should be able to handle it though.

As a result we should also move this up as it belongs with the other `StepVerifier.Step` return type rules.

Reply:
Should've replied to this comment :) https://github.com/PicnicSupermarket/error-prone-support/pull/617#issuecomment-1543404265